server/services/arduino/api/arduino.controller.js

```diff
@@ -0,0 +1,54 @@
+const asyncMiddleware = require('../../../api/middlewares/asyncMiddleware');
+
+module.exports = function ArduinoController(arduinoManager) {
+  /**
+   * @api {post} /api/v1/service/arduin/listen Listen
+   * @apiName listen
+   * @apiGroup Arduino
+   */
+  async function listen(req, res) {
+    arduinoManager.listen(req.body);
+    res.json({
+      success: true,
+    });
+  }
+
+  /**
+   * @api {post} /api/v1/service/arduin/configure Configure
+   * @apiName configure
+   * @apiGroup Arduino
+   */
+  async function configure(req, res) {
+    arduinoManager.configure(req.body);
+    res.json({
+      success: true,
+    });
+  }
+
+  /**
+   * @api {post} /api/v1/service/arduin/setup Setup
+   * @apiName setup
+   * @apiGroup Arduino
+   */
+  async function setup(req, res) {
+    arduinoManager.setup(req.body);
```

Comment:
From what I saw, this function never rejects any errors

How is the user going to know when something went wrong? 

Error handling is very important and you shouldn't `res.json({ success: true});` all the time. If something is wrong, reject a nice error message and in the UI display an explicit error message

Reply:
I modified setup.js script to return a json `{ success: true}` or `{ success: false}` to the front.

Reply:
This is not what I said! You should reject errors, not resolves with { success: false}. When something is wrong, the HTTP code should not be "200 ok" but should be relative to the error.

This is an example in the Philips Hue library : https://github.com/GladysAssistant/Gladys/blob/master/server/services/philips-hue/lib/light/light.setValue.js

When something is wrong, we throw a clear error which will results in a "404" error, clear for the client. 

We provide lots of errors in `utils/coreErrors` and `utils/httpErrors`. Let me know if you need more help :)